### PR TITLE
Pull requests to update population-city dataset

### PR DESCRIPTION
**Pull request by judeleonard**
@anuveyatsu could please review my PR for the issue #6

### CODE REVIEW

File: .github/workflows/update.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Population-city pipeline
+
+on:
+  push:
+    branches:
+      - master
+      - population_city
```
Comment (anuveyatsu) [inline .github/workflows/update.yaml]: @judeleonard i don't think we need the update job to run on any branches other than master

File: scripts/scraper.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from bs4 import BeautifulSoup
```
Comment (anuveyatsu) [inline scripts/scraper.py:6]: @judeleonard where are you using it?

File: .github/workflows/update.yaml (hunk 1 above)
Comment (judeleonard) [reply .github/workflows/update.yaml]: Updated

**Review by anuveyatsu — COMMENTED** [review]
@judeleonard am I correct that you are parsing the HTML table on the page? Do you think it is the right approach vs downloading the data in CSV?

File: scripts/scraper.py (hunk 2 above)
Comment (judeleonard) [reply scripts/scraper.py:6]: https://github.com/judeleonard/population-city/blob/population_city/scripts/scraper.py#L1

here

**Comment by judeleonard** [comment]
Based on the initial task to update the data, I thought about fetching first table where the update happened and then use that to update each of the file in the data.

**Comment by judeleonard** [comment]
Hi @anuveyatsu please what is the update on this task?

However, I forgot also to mention that the API endpoint to the UN website for fetching the data was blocked due to an SSL certification issue, so I settled for selenium for this task. Could it be possible to assign me another task if that is possible? I would be more than happy to pick that up. Thanks!

File: scripts/scraper.py (hunk 2 above)
Comment (anuveyatsu) [reply scripts/scraper.py:6]: @judeleonard  you imported `BeautifulSoup` but I couldn't find where are you using it.

**Comment by anuveyatsu** [comment]
@judeleonard I'm not sure about "the API endpoint to the UN website for fetching the data was blocked due to an SSL certification issue", could you elaborate on that? We normally wouldn't fetch HTML pages if we can access the API. My initial comment was also about that - why would one parse HTML when there is a clearer way to fetch data?

Also, this PR's title is not very descriptive for me.

**Comment by judeleonard** [comment]
@anuveyatsu  So after checking through the website I initially thought I could easily extract the link to the download button on the webpage but that was not the case. So I used their API documentation to understand how it can be used with the request library to retrieve the data. Below is a sample of my code that throws SSL error each time I make a request to get the data.

```python
import requests
from io import StringIO
import pandas as pd

url = "https://data.un.org/ws/rest/data/DF_UNData_UNFCC"
headers = {"Accept": "text/csv"}
response = requests.get(url, headers=headers)
data = response.json()
```

File: scripts/scraper.py (hunk 2 above)
Comment (judeleonard) [reply scripts/scraper.py:6]: The beautifulSoup is used here https://github.com/judeleonard/population-city/blob/6f193c14dc59db3af4e6b78971b1202bb1d0fc9b/scripts/scraper.py#L39 as a flavour for reading the HTML table

**Pull request closed without merging**